alacritty_terminal/src/input.rs

```diff
@@ -911,54 +908,36 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
     /// for its action to be executed.
     ///
     /// Return true if chars should be suppressed.
-    fn process_key_bindings(&mut self, input: KeyboardInput) -> bool {
-        self.key_bindings.iter().filter_map(|binding| {
-            let is_triggered = match binding.trigger {
-                Key::Scancode(_) => binding.is_triggered_by(
-                    *self.ctx.terminal().mode(),
-                    input.modifiers,
-                    &Key::Scancode(input.scancode),
-                    false,
-                ),
-                _ => {
-                    if let Some(key) = input.virtual_keycode {
-                        let key = Key::from_glutin_input(key);
-                        binding.is_triggered_by(
-                            *self.ctx.terminal().mode(),
-                            input.modifiers,
-                            &key,
-                            false,
-                        )
-                    } else {
-                        false
-                    }
-                },
-            };
+    fn process_key_bindings(&mut self, input: KeyboardInput) {
+        let mode = *self.ctx.terminal().mode();
+
+        *self.ctx.suppress_chars() = self
+            .key_bindings
+            .iter()
+            .filter(|binding| {
+                let key = match (binding.trigger, input.virtual_keycode) {
+                    (Key::Scancode(_), _) => Key::Scancode(input.scancode),
+                    (_, Some(key)) => Key::from_glutin_input(key),
+                    _ => return false,
+                };
 
-            if is_triggered {
+                binding.is_triggered_by(mode, input.modifiers, &key, false)
+            })
+            .fold(None, |suppress_chars, binding| {
                 // Binding was triggered; run the action
-                Some(binding.execute(&mut self.ctx, false))
-            } else {
-                None
-            }
-        }).fold(None, |should_suppress_char, suppress_char| {
-            // Suppress char by default as soon as a binding has been triggered
-            let should_suppress_char = should_suppress_char.unwrap_or(true);
-            Some(should_suppress_char && suppress_char)
-        }).unwrap_or_else(|| {
-            // Do not suppress char if no bindings were triggered
-            false
-        })
+                binding.execute(&mut self.ctx, false);
+
+                // Don't suppress when there has been a `ReceiveChar` action
+                Some(suppress_chars.unwrap_or(true) && binding.action != Action::ReceiveChar)
+            })
+            .unwrap_or(false);
```

Comment:
In my previous commit I added a comment for this to explain this `false`. If you don't feel that it's necessary, just ignore.

Reply:
Don't think there's anything wrong with a comment on this, definitely makes some sense, feel free to add it back!